### PR TITLE
feat: implement complete moon destruction mission system

### LINKED ISSUE #835: Moon Destruction Mission (Mission Type 9)
Add full support for Moon Destruction using Deathstars (RIPs), including mission rules, destruction chances, RIP loss chances, and redirection of fleets after a moon’s removal.

1. New Fleet Mission: Destroy
	•	Fleet mission available only to Deathstars.
	•	Fleets on a Destroy mission:
	•	Cannot join or form ACS groups.
	•	Must first fight any defending fleets/defenses on the moon.
	•	Only if this battle is won, the destruction attempt occurs.

2. Moon Destruction Logic
	•	Apply standard OGame formulas:
	•	Chance to destroy moon based on moon diameter and number of Deathstars.
	•	Chance to lose each Death Star during the attempt.

Formula:
`Moon Destruction Chance = (100 − √(Moon Diameter in km)) * √(Number of Death Stars)`

Formula for guaranteed Moon Destruction:

`Minimum RIPs = ( 1 / (1 − √(Moon Diameter in km / 100)) )²`

Deathstar Loss Chance:

`Death Star Loss Chance = √(Moon Diameter in km / 2)`

3. Destroyed Moon Effects
	•	No debris field is generated:
	•	Not from the destroyed moon.
	•	Not from RIPs lost during the attempt.
	•	Fleets targeting a destroyed moon are redirected to the associated planet, making them phalanx-visible and interceptable.

### PR DESCRIPTION
**Pull request by nguyenthanhan**
## Description

Implements moon destruction feature - allows players to use Deathstars to destroy enemy moons.

- Players send Deathstars on a Destroy mission to enemy moons
- Combat occurs first; if the attacker wins, the destruction attempt is calculated using OGame formulas
- Successful destruction removes the moon and redirects incoming fleets to the parent planet
- Includes comprehensive tests for formulas, combat integration, and edge cases

### Type of Change:
- [x] Feature enhancement

## Related Issues

Fixes #835

## Checklist

- [x] **Code Standards:** Code adheres to PSR-12 coding standards. Verified with Laravel Pint.
- [x] **Static Analysis:** Code passes PHPStan static code analysis.
- [x] **Testing:**
  - [x] Unit tests for destruction formulas and mission validation
  - [x] Feature tests for complete mission flow including edge cases
  - [x] Tests successfully run locally.
- [x] **CSS & JS Build:** N/A - No frontend changes
- [x] **Documentation:** In code

### CODE REVIEW

File: app/GameMissions/MoonDestructionMission.php [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+                $deathstarCount = $unit->amount;
+                break;
+            }
+        }
+
+        $destructionChance = $this->calculateMoonDestructionChance($moonDiameter, $deathstarCount);
+        $lossChance = $this->calculateDeathstarLossChance($moonDiameter);
+
+        $destructionRoll = random_int(0, 100);
+        $moonDestroyed = $destructionRoll <= $destructionChance;
```
Comment (Geda173) [inline app/GameMissions/MoonDestructionMission.php:192]: Using` random_int(0, 100)` gives 101 possible values. A 50% chance would succeed on 0-50 (51 values), making it ~50.5%. Consider using `random_int(1, 100)` with `<=`, or `random_int(0, 99)` with `<` for precise percentages.

File: app/GameMissions/MoonDestructionMission.php [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+        $deathstarsLost = 0;
```
Comment (Geda173) [inline app/GameMissions/MoonDestructionMission.php]: The official rules state this is a single roll for the entire fleet: either ALL Deathstars are destroyed or NONE are. This rolls per Deathstar, allowing partial losses (e.g., losing 3 out of 10 Deathstars). According to the wiki, it should be all-or-nothing:

`$lossRoll = random_int(0, 100);
$allDeathstarsLost = $lossRoll <= $lossChance;`

I marked it wrong in the file itself, I am talking about this whole block:

`$deathstarsLost = 0;
for ($i = 0; $i < $deathstarCount; $i++) {
    $lossRoll = random_int(0, 100);
    if ($lossRoll <= $lossChance) {
        $deathstarsLost++;
    }
}`

**Comment by Geda173** [comment]
Hi @nguyenthanhan,

great work. I have added some comments in the code where I think we may need a second pass. Let me know what you think!

**Comment by anngth** [comment]
Hi @Geda173, I appreciate your review! I agree with both points and will implement the fixes shortly.

**Comment by lanedirt** [comment]
Thanks for the work @nguyenthanhan and thanks for the review @Geda173. LGTM, merging!
